eng/targets/ResolveReferences.targets

```diff
@@ -74,16 +73,33 @@
       <IsSharedSource Condition="'%(IsSharedSource)' == '' AND $([System.String]::new('%(Identity)').EndsWith('.Sources'))">true</IsSharedSource>
     </Reference>
 
+    <!-- Packages which are implicitly defined by the .NET Core SDK. -->
     <_ImplicitPackageReference Include="@(PackageReference->WithMetadataValue('IsImplicitlyDefined', 'true'))" />
-    <_ExplicitPackageReference Include="@(PackageReference)" Exclude="@(_ImplicitPackageReference)" />
+    <!-- Capture a list of references which were set explicitly in the project. -->
+    <_AllowedExplicitPackageReference Include="@(PackageReference->WithMetadataValue('AllowExplicitReference', 'true'))" />
+    <_AllowedExplicitPackageReference Include="FSharp.Core" Condition="'$(MSBuildProjectExtension)' == '.fsproj'" />
```

Comment:
Do we know if there might be any other literal/explicit references like this one (now or in the future)?

Reply:
Actually I think this one wouldn't be needed. I copied changes from the AspNetCore but I don't think this one applies.